lms/services/application_instance_getter.py

```diff
@@ -0,0 +1,117 @@
+from Crypto.Cipher import AES
+from sqlalchemy.orm.exc import NoResultFound
+
+from lms.services import ConsumerKeyError
+from lms.models import ApplicationInstance
+
+
+class ApplicationInstanceGetter:
```

Comment:
`ApplicationInstanceGetter` doesn't sound like the best name to me since this service doesn't return `ApplicationInstance` objects, rather it returns information about them. `ApplicationInstanceInfo`? It might also be useful to briefly restate here what an "application instance" is (its an "install of the app within an LMS"?)

Reply:
I dunno. If we ever do have some code that does need to get actual `ApplicationInstance` objects, then this service is where the method for getting those would go. And I think `ApplicationInstanceGetter` sort of makes since now anyway since it's getting (properties from) application instances. I think I like having "getter" in there as it fits with the noun-verb naming style for services, like group fetcher, user creator, etc.

Reply:
Mmm ... I'm still not convinced but not strongly enough to object to merging.